format.md

```diff
@@ -5,9 +5,63 @@ can be reasonably confident that they have interpreted the source
 data in the same way - at least for the period of time covered by
 the tool.
 
-Current format
+Overall format
+====
+
+The entire file is a text file encoded using UTF-8, with no leading
+byte-order mark. Where the description mentions a line break, this
+is a Unix-style line break: the U+000A (`\n`) character.
+
+The file consists of an optional header section, follow by a body.
+
+Header section (optional)
+====
+
+This is simply a list of key/value pairs, in the form
+
+    *key* `:` *value*
+    
+followed by an empty line to indicate that the following line is the
+start of the body.
+
+The keys and values may contain (broadly) any letters, numbers,
+punctuation, although the key must not contain a colon (`:`).
+Separator characters **must not** occur in keys or values.
+(Future versions of this format may nail down the set of characters
+more thoroughly.)
+
+The order of headers is not significant.
+
+Leading and trailing whitespace is insignificant for headers, so the
+following header lines are equivalent:
+
+    HeaderKey:HeaderValue
+      HeaderKey  :  HeaderValue
+      
+Common headers
 ----
 
+The intention of the headers section is to provide metadata about
+the body section. A generator may produce any headers it deems
```

Comment:
Nope. I was tempted to make Format mandatory (and first) to indicate the presence of a header section at all, but as the first line of the body won't include a colon, it's easy to detect the presence of headers anyway.